src/cladetime/util/reference.py

```diff
@@ -70,20 +70,61 @@ def _get_s3_object_url(bucket_name: str, object_key: str, date: datetime) -> Tup
     return version_id, version_url
 
 
-def get_nextclade_dataset(
+def _run_nextclade_cli(
+    nextclade_cli_version: str, nextclade_command: list[str], output_file: Path, input_files: list[Path] | None = None
+) -> Path:
+    """Invoke Nextclade CLI commands via Docker."""
+
+    try:
+        client = docker.from_env()
+    except DockerException as err:
+        logger.error("Error creating py-docker client", error=err)
+        raise NextcladeNotAvailableError(
+            "Unable to create client for Nextstrain CLI. Is Docker installed and running?"
+        ) from err
+
+    output_path = output_file.parent
+    volumes = {str(output_path): {"bind": "/data/", "mode": "rw"}}
+
+    # if the nextclade command requires input files, add those to the volumes
+    # dictionary so they can be mounted in the Docker container
+    if input_files:
+        for file in input_files:
+            volumes[str(file)] = {"bind": f"/data/{file.name}", "mode": "rw"}
```

Comment:
Question about Docker volumes: if we mount `output_path` as `/data/` and then mount other files into `/data/{file.name}`, I assume those other files don't end up living in `output_path` outside of Docker?

Reply:
Good question--in fact, they do! The unit tests run against temp directories, so that wasn't immediately obvious.